src/plugins/field_formats/common/utils/highlight/highlight_request.ts

```diff
@@ -20,5 +20,6 @@ export function getHighlightRequest(shouldHighlight: boolean) {
       '*': {},
     },
     fragment_size: FRAGMENT_SIZE,
+    max_analyzed_offset: maxAnalyzedOffset,
```

Comment:
I'm wondering if there is a use case for exposing this as a setting other than to avoid the error?

If there is a use case for setting `max_analyzed_offset` less than [`index.highlight.max_analyzed_offset`](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-analyzed-offset) I think this is fine.

If there is no common use case, what about just setting this to a _really_ large value? Something like `Number.MAX_SAFE_INTEGER`. Because so long as the value is positive and greater than `index.highlight.max_analyzed_offset`, this would never be an issue.

Reply:
the error occurs when `max_analyzed_offset` is *LAGGER* than `index.highlight.max_analyzed_offset` AND text value is larger than `index.highlight.max_analyzed_offset`